tensorbay/cli/branch.py

```diff
@@ -4,38 +4,43 @@
 #
 
 """Implementation of gas branch."""
-
 import click
 
 from ..client.gas import DatasetClientType
+from ..exception import InternalServerError, StatusError, TensorBayException
 from .tbrn import TBRN, TBRNType
-from .utility import ContextInfo, error, get_dataset_client, get_gas, shorten
+from .utility import ContextInfo, error, get_dataset_client, get_gas, log_response, shorten
 
 
 def _implement_branch(
     obj: ContextInfo, tbrn: str, name: str, verbose: bool, is_delete: bool
 ) -> None:
-    info = TBRN(tbrn=tbrn)
-    if info.type != TBRNType.DATASET:
-        error(f'To operate a branch, "{info}" must be a dataset')
-
-    gas = get_gas(*obj)
-    dataset_client = get_dataset_client(gas, info)
+    try:
+        info = TBRN(tbrn=tbrn)
+        if info.type != TBRNType.DATASET:
+            error(f'To operate a branch, "{info}" must be a dataset')
 
-    if is_delete:
-        _delete_branch(dataset_client, info)
-        return
+        gas = get_gas(*obj)
+        dataset_client = get_dataset_client(gas, info)
 
-    if name:
-        _create_branch(dataset_client, name)
-    else:
-        _list_branches(dataset_client, verbose)
+        if is_delete:
+            _delete_branch(dataset_client, info)
+            return
 
+        if name:
+            _create_branch(dataset_client, name)
+        else:
+            _list_branches(dataset_client, verbose)
 
-def _create_branch(dataset_client: DatasetClientType, name: str) -> None:
-    if dataset_client.status.is_draft:
+    except StatusError:
```

Comment:
Keep the original check for commit id and draft number for now, and do not catch the `StatusError. Deal with it after unifying the information of `StatusError` later.